pkg/controller/operators/catalog/operator.go

```diff
@@ -976,22 +976,28 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		// not-satisfiable error
 		if _, ok := err.(solver.NotSatisfiable); ok {
 			logger.WithError(err).Debug("resolution failed")
-			updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
+			subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
+			_, updateErr := o.updateSubscriptionStatuses(subs)
 			if updateErr != nil {
 				logger.WithError(updateErr).Debug("failed to update subs conditions")
+				return updateErr
 			}
 			return nil
 		}
-		updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
+		subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
+		_, updateErr := o.updateSubscriptionStatuses(subs)
 		if updateErr != nil {
 			logger.WithError(updateErr).Debug("failed to update subs conditions")
+			return updateErr
 		}
 		return err
 	} else {
-		updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "", "", false)
+		subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "", "", false)
+		_, updateErr := o.updateSubscriptionStatuses(subs)
```

Comment:
Does it make sense to exclude from `subs` any Subscriptions in `updatedSubs` that are going to be updated again on L1033?

Reply:
@benluddy here we call `setSubsCond` and then `updateSubscriptionStatuses`, which results in the subscriptions being updated on cluster.
The call to `setSubsCond` for `updateSubs` again on L1033 is just to make sure `updatedSubs` are in sync with all `subs`, otherwise
1. the subscriptions get updated in L996
2. `updateSubs` however do not have the recent picture of the subscriptions (i.e the condition struct for `updatedSubs` is still the old condition)
3. The subscriptions are updated again when `updateSubs` are updated, i.e the work done in step 1 is overriden for `updatedSubs`

Given that, if the suggestion is to only update the subscriptions in L996 that are not in `updatedSubs`, the update to the `updateSubs` only happen after this in L1033, before which the function could return in multiple places if it encounters an error. That'd leave us in a state where only some subscriptions are updated, while others are not. 

It makes sense to update all the subscriptions here.

Reply:
On second thought, reducing the number of times we update the subscription statues sounds like a good idea. Added a new commit to do that.